src/memory/paging/mod.rs

```diff
@@ -363,7 +363,9 @@ pub fn remap_the_kernel<FA>(active_table: &mut ActivePageTable,
 
         for frame in Frame::range_inclusive(multiboot_start, multiboot_end) {
             let new_page = Page::containing_address(frame.start_address() + KERNEL_BASE);
-            mapper.map_to(new_page, frame, PRESENT, &mut allocator);
+            // if we have already mapped this page, it must have been
+            // already mapped when we mapped the elf sections.
+            let _ = mapper.map_to(new_page, frame, PRESENT, &mut allocator);
```

Comment:
I would check if the already mapped page is the same as the one we wish to map to. But it's not really needed.